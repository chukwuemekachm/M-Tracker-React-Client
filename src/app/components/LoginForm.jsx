import React, { Component } from 'react';
import PropTypes from 'prop-types';

class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(event) {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }

  onSubmit(event) {
    event.preventDefault();

    const { userLoginRequest, history } = this.props;
    userLoginRequest(this.state, history, 'login');
  }

  render() {
    const {
      email, password,
    } = this.state;
    return (
      <form onSubmit={this.onSubmit}>
        <label htmlFor="email" className="ch-label">
          Email Address
          {' '}
          <span className="ch-require">
            *
          </span>
          <input
            type="email"
            name="email"
            className="ch-input"
            required
            placeholder="Email Address"
            value={email}
            onChange={this.onChange}
          />
        </label>
        <label htmlFor="password" className="ch-label">
          Password
          {' '}
          <span className="ch-require">
            *
          </span>
          <input
            type="password"
            name="password"
            className="ch-input"
            required
            placeholder="Password"
            value={password}
            onChange={this.onChange}
          />
        </label>
        <button type="submit" className="ch-btn-flat">
          Login
        </button>
      </form>
    );
  }
}
LoginForm.propTypes = {
  userLoginRequest: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};

export default LoginForm;
