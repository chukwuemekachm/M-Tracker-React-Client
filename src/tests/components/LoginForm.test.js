import React from 'react';
import { mount } from 'enzyme';
import LoginForm from '../../app/components/LoginForm';

const userLoginRequest = jest.fn();
const history = {};
describe('Login form component', () => {
  it('should render without errors and submit the form', () => {
    const wrapper = mount(<LoginForm
      userLoginRequest={userLoginRequest}
      history={history}
    />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('[name="email"]').simulate('change', {
      target: { value: 'john@gmail.com' },
    });
    wrapper.find('[name="password"]').simulate('change', {
      target: { value: 'Password' },
    });
    wrapper.find('form').simulate('submit');
  });
});
