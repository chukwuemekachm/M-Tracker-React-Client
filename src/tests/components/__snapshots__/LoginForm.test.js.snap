// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form component should render without errors and submit the form 1`] = `
<LoginForm
  history={Object {}}
  userLoginRequest={[MockFunction]}
>
  <form
    onSubmit={[Function]}
  >
    <label
      className="ch-label"
      htmlFor="email"
    >
      Email Address
       
      <span
        className="ch-require"
      >
        *
      </span>
      <input
        className="ch-input"
        name="email"
        onChange={[Function]}
        placeholder="Email Address"
        required={true}
        type="email"
        value=""
      />
    </label>
    <label
      className="ch-label"
      htmlFor="password"
    >
      Password
       
      <span
        className="ch-require"
      >
        *
      </span>
      <input
        className="ch-input"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </label>
    <button
      className="ch-btn-flat"
      type="submit"
    >
      Login
    </button>
  </form>
</LoginForm>
`;
